homeassistant/components/tractive/__init__.py

```diff
@@ -0,0 +1,155 @@
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+from .const import (
+    DOMAIN,
+    RECONNECT_INTERVAL,
+    SERVER_AVAILABLE,
+    SERVER_UNAVAILABLE,
+    TRACKER_HARDWARE_STATUS_UPDATED,
+    TRACKER_POSITION_UPDATED,
+)
+
+PLATFORMS = ["device_tracker"]
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up tractive from a config entry."""
+    data = entry.data
+
+    if DOMAIN not in hass.data:
+        hass.data[DOMAIN] = {}
+
+    client = aiotractive.Tractive(data[CONF_EMAIL], data[CONF_PASSWORD])
+    try:
+        creds = await client.authenticate()
+    except aiotractive.exceptions.TractiveError as error:
+        await client.close()
+        raise ConfigEntryNotReady from error
+
+    tractive = TractiveClient(hass, client, creds["user_id"])
+    tractive.subscribe()
+
+    hass.data[DOMAIN][entry.entry_id] = tractive
+
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+
+    async def cancel_listen_task(_):
+        await tractive.unsubscribe()
+
+    entry.async_on_unload(
+        hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, cancel_listen_task)
+    )
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+    if unload_ok:
+        tractive = hass.data[DOMAIN].pop(entry.entry_id)
+        await tractive.unsubscribe()
+    return unload_ok
+
+
+class TractiveClient:
+    """A Tractive client."""
+
+    def __init__(self, hass, client, user_id):
+        """Initialize the client."""
+        self._hass = hass
+        self._client = client
+        self._user_id = user_id
+        self._listen_task = None
+
+    @property
+    def user_id(self):
+        """Return user id."""
+        return self._user_id
+
+    async def trackable_objects(self):
+        """Get list of trackable objects."""
+        return await self._client.trackable_objects()
+
+    def tracker(self, tracker_id):
+        """Get tracker by id."""
+        return self._client.tracker(tracker_id)
+
+    def subscribe(self):
+        """Start event listener coroutine."""
+        self._listen_task = asyncio.create_task(self._listen())
+
+    async def unsubscribe(self):
+        """Stop event listener coroutine."""
+        if self._listen_task:
+            self._listen_task.cancel()
+        await self._client.close()
+
+    async def _listen(self):
+        server_was_unavailable = False
+        while True:
+            try:
+                async for event in self._client.events():
+                    if server_was_unavailable:
+                        _LOGGER.debug("Tractive is back online")
+                        async_dispatcher_send(
+                            self._hass, f"{SERVER_AVAILABLE}-{self._user_id}"
```

Comment:
You do not listen for this event?
Should set available property for Entiy to False